.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,58 @@
+name: pull-request
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+    - source/**
+    - clik/**
+    - modules/**
+    - examples/**
+    - cmake/**
+    - hal/**
+    - .github/actions/do_build_ock/**
+    - .github/actions/setup_ubuntu_build/**
+    - .github/workflows/run_pr_tests.yml
+    - .github/workflows/pull-request.yml
+    - CMakeLists.txt
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+jobs:
+
+############### JOB   mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz:
+
+  mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 60
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4.1.0
+      with:
+        path: code
+    - run: cp -R code/.github .github
+    - name: setup-ubuntu
+      uses: ./.github/actions/setup_ubuntu_build
+      with:
+        llvm_version: '18'
+        llvm_build_type: RelAssert
+    - run: echo WORKSPACE is $GITHUB_WORKSPACE && echo PWD is `pwd` && ls -al
+    - name: build ock
+      uses: ./.github/actions/do_build_ock
+      with:
+        path: $GITHUB_WORKSPACE/code
+        install_dir: $GITHUB_WORKSPACE/install
+        build_targets: check-ock-UnitCL-group-vecz
+        mux_targets_enable: riscv
+        external_compiler_dirs: ${{ github.workspace }}/code/examples/refsi/refsi_m1/compiler/refsi_m1
+        mux_compilers_enable: riscv
+        riscv_enabled: ON
+        enable_rvv_scalable_vecz_check: ON
+        enable_rvv_scalable_vp_vecz_check: ON
+        use_linker: gold
```

Comment:
if this is better, lets just make it the default.

Reply:
All of the jobs I've looked at so far do use it.
What does "better" mean? How is it measured?

Reply:
I believe the gold linker is faster. Perhaps we can review the defaults separately.

Reply:
I'll add that to the "To Do"list in the ticket and resolve this conversation.